cai_bi/models/dataconsumption/financial/timesheet_entry_aggregates/dim_timesheet_entry_aggregates.sql

```diff
@@ -71,297 +75,889 @@ WITH
+                cast(sum(te.cost_amount_usd) as number(38,2)) as amt_cost_usd,
+                cast(sum(te.cost_amount_cola_usd) as number(38,2)) as amt_cost_cola_usd,
+                cast(div0(amt_bill_usd, hours) as number(38,2)) as avg_bill_rate_usd,
+                cast(div0(amt_cost_usd, hours) as number(38,2)) as avg_cost_rate_usd,
+                cast(div0(amt_cost_cola_usd, hours) as number(38,2)) as avg_cost_rate_cola_usd
             from base_timesheet_entry as te
             where te.task_name in (select phase_code from time_type_phase_codes where time_type = 'pto')
             group by all
             union (
-                select key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year, hours,  avg_rate_employee, avg_rate_usd, amount_employee, amount_usd
+                select key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year, hours, hours_billable, currency_code_entity, amt_bill_entity, amt_cost_entity, 
+                    amt_cost_cola_entity, avg_bill_rate_entity, avg_cost_rate_entity, avg_cost_rate_cola_entity, currency_code_usd, amt_bill_usd, amt_cost_usd, amt_cost_cola_usd, avg_bill_rate_usd, 
+                    avg_cost_rate_usd, avg_cost_rate_cola_usd
                 from blank_data
             )
         ) as p
         group by all
     ),
 
+    -- ************************************************************************************************************************************************
+    -- PTO_DATA_PROJECT_ENTITY: get the sum of all pto data grouped by project entity limiting to only pto entries
+    pto_data_project_entity as (
+        select 'PTO' as type, 'Project' as entity_grouping, 103 as type_sort, key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year,
+            sum(hours) as hours,
+            sum(hours_billable) as hours_billable,
+            
+            -- get aggregates for entity
+            currency_code_entity,
+            sum(amt_bill_entity) as amt_bill_entity,
+            sum(amt_cost_entity) as amt_cost_entity,
+            sum(amt_cost_cola_entity) as amt_cost_cola_entity,
+            sum(avg_bill_rate_entity) as avg_bill_rate_entity,
+            sum(avg_cost_rate_entity) as avg_cost_rate_entity,
+            sum(avg_cost_rate_cola_entity) as avg_cost_rate_cola_entity,
+
+            -- get aggregates for USD
+            'USD' as currency_code_usd,
+            sum(amt_bill_usd) as amt_bill_usd,
+            sum(amt_cost_usd) as amt_cost_usd,
+            sum(amt_cost_cola_usd) as amt_cost_cola_usd,
+            sum(avg_bill_rate_usd) as avg_bill_rate_usd,
+            sum(avg_cost_rate_usd) as avg_cost_rate_usd,
+            sum(avg_cost_rate_cola_usd) as avg_cost_rate_cola_usd
+        from (
+            select
+                te.key_employee, key_entity_project as key_entity, te.employee_id, te.date_group_id, te.date_group_type_id, te.date_group_year,
```

Comment:
Each of these columns should all be on a new line so it's consistent. It is hard to read as one line